### PR TITLE
Make notes fields handle vertical content

### PR DESCRIPTION
**Pull request by jazeee**
The notes fields for spells and elsewhere should expand vertically. #5

Also, fix some alignment and margin issues.

## Old:

image
image

## New:

image

image

### CODE REVIEW

File: bin/main.css [hunk 1]
```diff
@@ -1107,13 +1103,6 @@ input.featCP {
   align-items: flex-end;
 }
 
-.spell-row > input ~ input,
-.spell-row > input ~ select,
-.spell-row > select ~ input {
-  /* Make consecutive buttons overlap */
-  margin-left: -1px;
```
Comment (jazeee) [inline bin/main.css:1114]: Negative margins are a bit of a pain.

File: ui/roll20_character_sheet_0.2.html [hunk 2]
```diff
@@ -956,7 +956,7 @@
         <!-- The next input puts whether we have a note into the
           value property, where CSS can see it -->
         <input name="attr_spellhasnote" class="spellhasnote" type="hidden">
-        <input name="attr_spellnote" class="spell-note" type="text">
+        <textarea name="attr_spellnote" class="spell-note"/>
```
Comment (jazeee) [inline ui/roll20_character_sheet_0.2.html]: Unsure if this is what #5 is asking for.

File: ui/spells/spells.scss [hunk 3]
```diff
@@ -22,22 +22,24 @@
...
   display: flex;
   align-items: center;
-}
 
-.spell-tabs>button[type="action"] {
-  border-bottom: 0px;
-}
+  >button[type="action"] {
+    border-bottom: 0px;
+  }
```
Comment (jazeee) [inline ui/spells/spells.scss:28]: I also started nesting classes. See end result in `main.css`
No difference from these changes.

Review without whitespace: https://github.com/epic-power-rpg/roll20/pull/23/files?diff=unified&w=1

File: ui/spells/spells.scss [hunk 4]
```diff
@@ -78,19 +80,22 @@
...
+  ~.spellduration,
+  ~.spellhasnote {
+    display: none;
+  }
 
-.isdiscipline[value="1"]~.spellhasnote+span.skillTP {
-  margin-left: 442px;
+  ~.spellhasnote+span.skillTP {
+    // This aligns the `Experience` column when the row is `isdispline`
+    margin-left: 448px;
```
Comment (jazeee) [inline ui/spells/spells.scss:97]: This is a new change. It aligns the D select with the other rows.

File: ui/spells/spells.scss [hunk 5]
```diff
@@ -100,42 +105,30 @@
...
 }
-
-.spell-note.spell-note.spell-note.spell-note {
-  text-align: left;
-  margin-top: -1px;
-}
-
 .arcane .spell-note {
-  margin-left: 90px;
+  margin-left: 40px;
```
Comment (jazeee) [inline ui/spells/spells.scss:110]: I felt this looked better.

File: ui/spells/spells.scss [hunk 6]
```diff
@@ -100,42 +105,30 @@
 .spell-note {
   flex-grow: 1;
   margin-bottom: 1px;
-  margin-left: 50px;
```
Comment (jazeee) [inline ui/spells/spells.scss:103]: For non-arcane, this seemed unneeded.

File: ui/spells/spells.scss [hunk 7]
```diff
@@ -100,42 +105,30 @@
...
 }
-
-.spell-note.spell-note.spell-note.spell-note {
-  text-align: left;
-  margin-top: -1px;
```
Comment (jazeee) [inline ui/spells/spells.scss:108]: Negative margins make the header not align with the fields.

File: ui/spells/spells.scss [hunk 8]
```diff
@@ -100,42 +105,30 @@
...
-  align-items: flex-end;
-}
-
-.spells div[data-groupname]>div {
-  display: flex;
-  flex-direction: column;
-  align-items: stretch;
-  /* and make rows overlap */
-  margin-top: -1px;
+.spells {
```
Comment (jazeee) [inline ui/spells/spells.scss:114]: No change, just nesting.

File: ui/spells/spells.scss [hunk 9]
```diff
@@ -100,42 +105,30 @@
...
   flex-direction: row;
   align-items: flex-end;
-}
-
-.spell-row>input~input,
-.spell-row>input~select,
-.spell-row>select~input {
-  /* Make consecutive buttons overlap */
-  margin-left: -1px;
-}
```
Comment (jazeee) [inline ui/spells/spells.scss:141]: Deleted this negative margin for the same reason.

File: ui/spells/spells.scss (hunk 8 above)
Comment (jazeee) [reply ui/spells/spells.scss:114]: Review without whitespace to see:
https://github.com/epic-power-rpg/roll20/pull/23/files?diff=unified&w=1

File: ui/spells/spells.scss [hunk 10]
```diff
@@ -78,19 +80,22 @@
...
+.isdiscipline[value="1"] {
+  ~.skill-roll,
+  ~.spellEP,
+  ~.spellcasttime,
+  ~.spelltype,
+  ~.spellrange,
+  ~.spelltarget,
+  ~.spellduration,
+  ~.spellhasnote {
+    display: none;
```
Comment (jazeee) [inline ui/spells/spells.scss:92]: No change, just nesting.

File: package.json [hunk 11]
```diff
@@ -23,7 +23,7 @@
     "lint:all": "yarn lint --ext .js .",
     "lint:fix-all": "yarn lint:all --fix",
     "build:all": "npm-run-all --print-name --parallel sass html",
-    "html": "echo 'TODO' > ./bin/main.html",
+    "html": "cat ./ui/roll20_character_sheet_0.2.html > ./bin/main.html",
```
Comment (jazeee) [inline package.json:26]: For now, we're just copying to `/bin`

**Review by johnlamping — APPROVED** [review]
This looks like a good improvement.
There are some small formatting tweaks that we can do later:
   There is a larger gap between a spell and its note than between adjacent spells.
   There is a larger gap between different info for a spell than between spells
   The left side of spells is darker, but not the left side of notes.

**Pull request merged**